src/commonMain/kotlin/fr/acinq/lightning/message/Postman.kt

```diff
@@ -0,0 +1,153 @@
+package fr.acinq.lightning.message
+
+import fr.acinq.bitcoin.ByteVector
+import fr.acinq.bitcoin.ByteVector32
+import fr.acinq.bitcoin.PrivateKey
+import fr.acinq.bitcoin.PublicKey
+import fr.acinq.lightning.Lightning.randomBytes32
+import fr.acinq.lightning.Lightning.randomKey
+import fr.acinq.lightning.crypto.RouteBlinding
+import fr.acinq.lightning.crypto.sphinx.Sphinx
+import fr.acinq.lightning.utils.Either
+import fr.acinq.lightning.utils.Either.Left
+import fr.acinq.lightning.utils.Either.Right
+import fr.acinq.lightning.wire.*
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.SendChannel
+import kotlinx.coroutines.withTimeoutOrNull
+import kotlin.math.max
+import kotlin.time.Duration
+
+class Postman(
```

Comment:
Since we're only supporting quite a small set of actions compared to eclair (sending and receiving onion messages in the context of bolt12 payments), and only plan on supporting very specific onion messages flows in the future on a case-by-case basis, I think we should optimize for readability and be much more explicit about what we are doing. I believe we only support the following scenarios:

- receiving an `onion_message` containing an `invoice_request`, to which we respond with an invoice if it matches one of our offers
- sending an `invoice_request`
- receiving an `onion_message` in response to one of our `invoice_request`, which can either:
  - contain a bolt 12 invoice that we'll want to pay
  - contain an `invoice_error`
- everything else should just be silently ignored

Is that correct? If that's the case, I think a clear API exposed by the `Postman` should look like this:

```kotlin
sealed class OnionMessageAction {
    data class SendMessage(val message: OnionMessage): OnionMessageAction()
    data class PayInvoice(val payment: SendPayment, val invoice: Bolt12Invoice): OnionMessageAction() 
    data class ReportPaymentFailure(val payment: SendPayment, val failure: PaymentFailure): OnionMessageAction()
}

sealed class PaymentFailure {
    data object InvoiceNotReceived: PaymentFailure()
    data class InvoiceError(val message: String): PaymentFailure()
}

class Postman(val privateKey: PrivateKey, val remoteNodeId: PublicKey) {
    // I believe that this function doesn't need to suspend, as we don't want to block waiting for the invoice.
    // We'll need to slightly modify the existing SendPayment class (from Peer.kt) to handle bolt 12 payments (maybe make it a sealed class).  
    fun requestInvoice(payment: SendPayment): List<OnionMessageAction> {
        // Create the onion message that sends the invoice_request.
        // Record this outgoing message in our subscribed map (which can probably be renamed pending_invoice_requests for clarity).
        // Return OnionMessageAction.SendMessage.
        TODO()
    }

    suspend fun receiveMessage(msg: OnionMessage): List<OnionMessageAction> {
        // Decrypt onion message.
        // If this is an invoice_request:
        //  - check that the offer matches an offer in our DB
        //  - if it does, generate an invoice, create an outgoing onion message using the reply path, and return OnionMessageAction.SendMessage
        //  - otherwise, just ignore and return listOf()
        // If this is a valid Bolt 12 invoice that can be found in our pending_invoice_requests map:
        //  - return OnionMessageAction.PayInvoice and remove from our map
        // If this is an invoice_error that can be found in our pending_invoice_requests map:
        //  - return OnionMessageAction.ReportPaymentFailure and remove from our map
        // Otherwise:
        //  - just ignore and return listOf()
        TODO()
    }

    fun checkInvoiceRequestTimeout(timeout: Duration): List<OnionMessageAction> {
        // For each request in pending_invoice_requests that was created more than <timeout> ago:
        //  - remove it from the map
        //  - return OnionMessageAction.ReportPaymentFailure with PaymentFailure.InvoiceNotReceived
        TODO()
    }
}
```